### PR TITLE
Updating version of xUnit-Performance-Api

### PR DESCRIPTION
**Pull request by jorive**
- This update contains a fix for this issue: https://github.com/Microsoft/xunit-performance/issues/230
- Consolidate the Api version in two places: `dependencies.prop` and `PerfHarness.csproj`
- Move performance log files to root directory where Jenkins can archive them

*NOTE: This change will reflect on some benchmarks improving their timings*

### CODE REVIEW

**Comment by jorive** [comment]
test ci please

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf
@dotnet-bot test Ubuntu14.04 x64 perf

File: tests/src/Common/PerfHarness/PerfHarness.csproj [hunk 1]
```diff
@@ -6,9 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="xunit.performance.api">
-      <Version>1.0.0-beta-build0006</Version>
-    </PackageReference>
+    <PackageReference Include="xunit.performance.api" Version="1.0.0-beta-build0007" />
```
Comment (wtgodbe) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: I don't think this works with all versions of msbuild - @weshaggard can you comment?

Comment (weshaggard) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: Correct this will only work if you load this project with the new msbuild 15 or msbuild core that comes with the SDK. If someone tries to load this project in VS2015 or msbuild <15 then it will blow-up. It might not be an issue for this particular project but something that needs to be kept in mind.

Comment (jorive) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: @weshaggard Is there a way to import `dir.props` and `dir.targets` in the new `csproj` file? I would like to have the version of the package in a central place. Is that a supported scenario?

Comment (weshaggard) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: Yes you can still do an import the same way as you used to be able to `<Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />`

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf
@dotnet-bot test Ubuntu14.04 x64 perf
test ci please

File: tests/src/Common/PerfHarness/PerfHarness.csproj (hunk 1 above)
Comment (jorive) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: If I add `Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />` to the `PerfHarness.csproj`, then I can restore, but I cannot publish. What am I missing?

```cmd
$ "%CORECLR_ROOT%Tools\dotnetcli\dotnet.exe" --info
.NET Command Line Tools (2.0.0-preview1-005724)

Product Information:
 Version:            2.0.0-preview1-005724
 Commit SHA-1 hash:  e391b5d5f3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   %CORECLR_ROOT%\Tools\dotnetcli\sdk\2.0.0-preview1-005724\

$ "%CORECLR_ROOT%\Tools\dotnetcli\dotnet.exe" restore "%CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj"
%CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj
  Restoring packages for %CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj...
  Lock file has not changed. Skipping lock file write. Path: %CORECLR_ROOT%\tests\src\Common\PerfHarness\obj\project.assets.json
  Restore completed in 280.64 ms for %CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj.

  NuGet Config files used:
      %CORECLR_ROOT%\tests\src\NuGet.Config
      %APPDATA%\NuGet\NuGet.Config
      C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

  Feeds used:
      https://dotnet.myget.org/F/dotnet-core/api/v3/index.json
      https://www.myget.org/F/nugetbuild/api/v3/index.json
      https://api.nuget.org/v3/index.json

$ "%CORECLR_ROOT%\Tools\dotnetcli\dotnet.exe" publish "%CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj" -c Release -o "%CORECLR_ROOT%\sandbox"

Microsoft (R) Build Engine version 15.2.93.5465
Copyright (C) Microsoft Corporation. All rights reserved.

%CORECLR_ROOT%\Tools\Microsoft.CSharp.Core.targets(106,11): error MSB4064: The "OverrideToolHost" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property. [%CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj]
%CORECLR_ROOT%\Tools\Microsoft.CSharp.Core.targets(67,5): error MSB4063: The "Csc" task could not be initialized with its input parameters.  [%CORECLR_ROOT%\tests\src\Common\PerfHarness\PerfHarness.csproj]
```

**Comment by jorive** [comment]
test ci please
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf
@dotnet-bot test Ubuntu14.04 x64 perf

**Comment by jorive** [comment]
@dotnet-bot help

**Comment by jorive** [comment]
@dotnet-bot test Ubuntu14.04 perf

**Comment by jorive** [comment]
test ci please
@dotnet-bot test Windows_NT perf
@dotnet-bot test Ubuntu14.04 perf

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf

File: tests/scripts/run-xunit-perf.cmd [hunk 2]
```diff
@@ -377,9 +384,6 @@ rem   Sends text to the console screen, no matter what (even when the script's
 rem   output is redirected). This can be useful to provide information on where
 rem   the script is executing.
 rem ****************************************************************************
-  if defined _debug (
```
Comment (dpodder) [inline tests/scripts/run-xunit-perf.cmd]: Could we delete the entire `:print_to_console` function? I don't think anybody uses it anymore, right?

File: tests/scripts/run-xunit-perf.cmd [hunk 3]
```diff
@@ -73,27 +71,37 @@ setlocal
...
-    call :run_cmd corerun.exe "%CORECLR_REPO%\sandbox\%BENCHNAME%.%TEST_FILE_EXT%" --perf:runid Perf 1>"%BENCHNAME_LOG_FILE_NAME%" 2>&1
+    set "LV_CMD=corerun.exe "%CORECLR_REPO%\sandbox\%BENCHNAME%.%TEST_FILE_EXT%" --perf:runid "%LV_RUNID%""
   ) else (
-    call :run_cmd %STABILITY_PREFIX% corerun.exe PerfHarness.dll "%CORECLR_REPO%\sandbox\%BENCHNAME%.%TEST_FILE_EXT%" --perf:runid Perf --perf:collect %COLLECTION_FLAGS% 1>"%BENCHNAME_LOG_FILE_NAME%" 2>&1
+    set "LV_CMD=%STABILITY_PREFIX% corerun.exe PerfHarness.dll "%CORECLR_REPO%\sandbox\%BENCHNAME%.%TEST_FILE_EXT%" --perf:runid "%LV_RUNID%" --perf:collect %COLLECTION_FLAGS%"
   )
 
+  echo/
+  echo/%USERNAME%@%COMPUTERNAME% "%CD%"
+  echo/[%DATE%][%TIME:~0,-3%] $ !LV_CMD!
```
Comment (dpodder) [inline tests/scripts/run-xunit-perf.cmd]: It looks like `:run_cmd` already prints this out. Doesn't this cause us to just print everything twice?

Comment (jorive) [inline tests/scripts/run-xunit-perf.cmd]: No in this case, because we are redirecting to a log file.

File: tests/scripts/run-xunit-perf.cmd (hunk 2 above)
Comment (jorive) [inline tests/scripts/run-xunit-perf.cmd]: I should be reusing that code. I'll fix it.

File: tests/scripts/run-xunit-perf.cmd (hunk 3 above)
Comment (dpodder) [inline tests/scripts/run-xunit-perf.cmd]: Ah... d'oh... I missed the obvious. Looks good.

File: tests/scripts/run-xunit-perf.cmd (hunk 2 above)
Comment (dpodder) [inline tests/scripts/run-xunit-perf.cmd]: 🆗

**Comment by jorive** [comment]
test ci please
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf
@dotnet-bot test Ubuntu14.04 perf

File: tests/scripts/run-xunit-perf.cmd (hunk 2 above)
Comment (jorive) [inline tests/scripts/run-xunit-perf.cmd]: fixed.

**Comment by jorive** [comment]
test ci please
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf
@dotnet-bot test Ubuntu14.04 perf

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf

**Comment by jorive** [comment]
@dotnet-bot help

**Comment by jorive** [comment]
@dotnet-bot test Ubuntu arm Cross Release Build

**Comment by jorive** [comment]
@dotnet-bot test Windows_NT x64 perf

**Comment by AndyAyersMS** [comment]
@jorive did you see performance shifts in some tests? Can you summarize?

**Comment by jorive** [comment]
Yes, I did see some performance shifts. Some benchmarks improved timings and some got slower. I'll send you links to the results.
